src/MultiCheck/MultiCheck.tsx

```diff
@@ -28,9 +30,74 @@ type Props = {
 }
 
 const MultiCheck: React.FunctionComponent<Props> = (props): JSX.Element => {
-  return <div className='MultiCheck'>
-    {/* TODO */}
-  </div>
+  /**
+   * State for whether ths "Select All" check box is checked
+   */
+  const [selectAllIsChecked, setSelectAllIsChecked] = useState(
+    props.values ? props.values?.length === props.options.length : false
+  );
+  /**
+   * State for whether each checkbox is checked
+   */
+  const [optionsAreChecked, setOptionsAreChecked] = useState(
+    props.options.map((o) => props.values?.includes(o.value))
+  );
+
+  /**
+   * Click listener
+   */
+  const handleSelectAllCheckBoxChanged = (isSelectAllChecked: boolean) => {
+    setOptionsAreChecked(optionsAreChecked.map(() => isSelectAllChecked));
+  }
+  const handleCheckBoxChanged = (index: number, isChecked: boolean) => {
+    setOptionsAreChecked(optionsAreChecked.map((o, i) => (i === index) ? isChecked : o));
+  }
+
+  /**
+   * When the state allCheckBoxsAreSelected is changed
+   */
+  useEffect(() => {
+    const allCheckBoxsAreSelected: boolean = optionsAreChecked.every((o) => o);
+    setSelectAllIsChecked(allCheckBoxsAreSelected);
+    // Update the selected values outside, passed in by props
+    props.onChange && props.onChange(props.options.filter((_, i) => optionsAreChecked[i]));
+  }, [optionsAreChecked])
+
+  return (
+    <div className='MultiCheck'>
+      <h2 className='labelName'>{props.label}</h2>
+      <div
+        className='checkBoxesContainer'
+        style={{ columns: props.columns ? props.columns : 1 }}>
+        <div>
+          <input
+            type='checkbox'
+            id='selectAllCheckBox'
+            name='selectAllCheckBox'
+            value={SELECT_ALL}
+            onChange={(e) => { handleSelectAllCheckBoxChanged(e.target.checked) }}
+            checked={selectAllIsChecked}
+          />
+          <label htmlFor='selectAllCheckBox'>{SELECT_ALL}</label>
+        </div>
+
+        {props.options.map((option, index) => (
+          <div key={index + 1}>
```

Comment:
how about
```suggestion
          <div key={`${option.label}-${option.value}`}>
```
or, if we decide to use index, why don't we just
```suggestion
          <div key={index}>
```